### PR TITLE
Adding Grid Engine

### PR DESCRIPTION
**Pull request by PatB93**

### CODE REVIEW

**Comment by lucas-brandt** [comment]
not ready for review?

File: engine/src/main/java/com/example/engine/MapGrid.kt [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package com.example.engine
+
+import java.util.*
+
+class MapGrid(mapWidth: Int, mapHeight: Int, gridSize: Int = 24, mapArray: Array<Array<SpriteType>>?) {
```
Comment (lucas-brandt) [inline engine/src/main/java/com/example/engine/MapGrid.kt]: Why make SpriteType nullable? Aren't we always going to want to guarantee a sprite shows up on the map?

Comment (PatB93) [reply engine/src/main/java/com/example/engine/MapGrid.kt]: That is in the case we don't have one pre-defined. This class will assume that if we don't give it a mapArray, there is no saved state. Then we go to `generateArray()`

File: engine/src/main/java/com/example/engine/Sprites.kt [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+enum class SpriteType {
+    GRASS,
+    DIRT_WET,
+    DIRT_DRY,
+    TREE,
+    BUILDING
+}
+
+abstract class Sprites(val type: SpriteType, val width: Int, val height: Int, val imageRes: Int)
+class GroundTile(type: SpriteType, width: Int = 24, height: Int = 24, imageRes: Int = R.drawable.ic_grass_tile) : Sprites(type, width, height, imageRes) {
```
Comment (lucas-brandt) [inline engine/src/main/java/com/example/engine/Sprites.kt]: 👾

File: engine/src/test/java/com/example/engine/MapGridTest.kt [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+import org.junit.Test
+
+class MapGridTest {
+
+    @MockK
+    lateinit var groundTile: GroundTile
+
+    @MockK
+    lateinit var treeTile: TreeTile
+
```
Comment (lucas-brandt) [inline engine/src/test/java/com/example/engine/MapGridTest.kt]: Where BuildingTile?

File: engine/src/main/java/com/example/engine/MapGrid.kt (hunk 1 above)
Comment (lucas-brandt) [reply engine/src/main/java/com/example/engine/MapGrid.kt]: Okay, got it.

File: engine/src/test/java/com/example/engine/MapGridTest.kt (hunk 3 above)
Comment (PatB93) [reply engine/src/test/java/com/example/engine/MapGridTest.kt]: Added to test

**Pull request merged**